### PR TITLE
Doctests for linalg

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/linalg/ldlt.jl [hunk 1]
```diff
@@ -20,6 +20,26 @@ convert(::Type{Factorization{T}}, F::LDLt{S,U}) where {T,S,U} = convert(LDLt{T,U
     ldltfact!(S::SymTridiagonal) -> LDLt
 
 Same as [`ldltfact`](@ref), but saves space by overwriting the input `A`, instead of creating a copy.
```
Comment (fredrikekre) [inline base/linalg/ldlt.jl]: Perhaps fix `A` -> `S` such that the signature and docstring matches.

File: base/linalg/ldlt.jl [hunk 2]
```diff
@@ -38,6 +58,31 @@ end
 Compute an `LDLt` factorization of a real symmetric tridiagonal matrix such that `A = L*Diagonal(d)*L'`
```
Comment (fredrikekre) [inline base/linalg/ldlt.jl]: Same thing a couple of times in this text.

File: base/linalg/generic.jl [hunk 3]
```diff
@@ -591,6 +612,33 @@ value `p = q/(q-1)`. They coincide at `p = q = 2`.
 The difference in norm between a vector space and its dual arises to preserve
 the relationship between duality and the inner product, and the result is
 consistent with the p-norm of `1 × n` matrix.
+
+# Examples
+```jldoctest
+julia> v = [1; im];
+
+julia> vc = v'
+1×2 RowVector{Complex{Int64},ConjArray{Complex{Int64},1,Array{Complex{Int64},1}}}:
```
Comment (fredrikekre) [inline base/linalg/generic.jl]: Perhaps suppress the output of this hideous type?

File: base/linalg/generic.jl [hunk 4]
```diff
@@ -591,6 +612,33 @@ value `p = q/(q-1)`. They coincide at `p = q = 2`.
...
+
+julia> norm(v, 1)
+2.0
+
+julia> norm(v, 2)
+1.4142135623730951
+
+julia> norm(v, Inf)
+1.0
+```
```
Comment (fredrikekre) [inline base/linalg/generic.jl]: Perhaps we can do something like 
```
norm(v, 1)
norm(vc, 1)
norm(v, 2)
norm(vc, 2)
norm(v, Inf)
norm(vc, Inf)
```
such that the vector and rowvector cases for same q are closer?

File: base/linalg/generic.jl [hunk 5]
```diff
@@ -451,6 +451,27 @@ end
...
+2
+
+julia> vecnorm(-2, Inf)
+2
+
+julia> vecnorm(-2, 2)
+2
+
+julia> vecnorm(-2, 1)
+2
```
Comment (fredrikekre) [inline base/linalg/generic.jl]: Perhaps order them as:
```julia
vecnorm( 2, 1)
vecnorm(-2, 1)
vecnorm( 2, 2)
vecnorm(-2, 2)
vecnorm( 2, Inf)
vecnorm(-2, Inf)
```

**Comment by kshyatt** [comment]
@fredrikekre thanks for the helpful feedback! look good to go if CI passes?

File: base/linalg/ldlt.jl (hunk 1 above)
Comment (fredrikekre) [reply base/linalg/ldlt.jl]: Perhaps I wasn't clear enough; what I meant was that in the docstring we use `A` but in the signature two rows above we call it `S`.

File: base/linalg/ldlt.jl (hunk 2 above)
Comment (fredrikekre) [reply base/linalg/ldlt.jl]: Same thing here, I will fix if you don't mind.

**Review by fredrikekre — APPROVED** [review]

File: base/linalg/ldlt.jl (hunk 1 above)
Comment (kshyatt) [reply base/linalg/ldlt.jl]: You were probably plenty clear, I'm just braindead on Friday afternoon 😛

**Comment by fredrikekre** [comment]
#23796 × 2

**Pull request merged**